src/Filters/coefficients.jl

```diff
@@ -179,13 +177,8 @@ function Base.:^(f::PolynomialRatio{D,T}, e::Integer) where {D,T}
     end
 end
 
-function _rev_zeropad(v::AbstractVector{T}, n, pad_len=n) where T
-    padded_v = zeros(T, length(v) + n)
-    return copyto!(padded_v, 1 + pad_len, Iterators.reverse(v))
-end
-
-coef_s(p::LaurentPolynomial) = _rev_zeropad(coeffs(p), firstindex(p), 0)
-coef_z(p::LaurentPolynomial) = _rev_zeropad(coeffs(p), -lastindex(p))
+coef_s(p::LaurentPolynomial{T}) where T = (n = firstindex(p);  append!(reverse!(coeffs(p)), zero(T) for _ in 1:n))
+coef_z(p::LaurentPolynomial{T}) where T = (n = -lastindex(p); prepend!(reverse!(coeffs(p)), zero(T) for _ in 1:n))
```

Comment:
`coeffs` acts as a view, so this would change `p`.
```suggestion
coef_s(p::LaurentPolynomial{T}) where T = (n = firstindex(p);  append!(reverse(coeffs(p)), zero(T) for _ in 1:n))
coef_z(p::LaurentPolynomial{T}) where T = (n = -lastindex(p); prepend!(reverse(coeffs(p)), zero(T) for _ in 1:n))
```

Reply:
Also, these lines are quite long. Either make these long-form functions or at least
```suggestion
coef_s(p::LaurentPolynomial{T}) where T =
    append!( reverse(coeffs(p)), zero(T) for _ in 1:firstindex(p))
coef_z(p::LaurentPolynomial{T}) where T =
    prepend!(reverse(coeffs(p)), zero(T) for _ in 1:-lastindex(p))
```

Reply:
```julia
julia> using Polynomials

julia> p = LaurentPolynomial([1, 2, 3, 4, 5], -1)
LaurentPolynomial(x⁻¹ + 2 + 3*x + 4*x² + 5*x³)

julia> reverse!(coeffs(p));

julia> p
LaurentPolynomial(5*x⁻¹ + 4 + 3*x + 2*x² + x³)
```

Reply:
Ah, sorry, that was Polynomials v3, with v4 it seems you are correct.